From 26958746c69a4560b8daa22f9f8ed9b88472ef6a Mon Sep 17 00:00:00 2001
From: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
Date: Mon, 30 Jun 2014 16:37:14 +0100
Subject: [PATCH 241/351] switch: arizona: fix runtime pm calls

Use pm_runtime_get_sync instead of pm_runtime_get so that we are
sure that the device has resumed and we can access the registers
immediately when we return from the call.

Use pm_runtime_mark_last_busy just after the last register was
accesed and before calling pm_runtime_put_autosuspend

Change-Id: Ica6a88ff4685a0c0f32acf798667256b2d9d4c21
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 83c2071..fa637e3 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -732,7 +732,7 @@ int arizona_hpdet_start(struct arizona_extcon_info *info)
 	}
 
 	/* Make sure we keep the device enabled during the measurement */
-	pm_runtime_get(info->dev);
+	pm_runtime_get_sync(info->dev);
 
 	arizona_extcon_do_magic(info, 0x4000);
 
@@ -799,6 +799,7 @@ void arizona_hpdet_stop(struct arizona_extcon_info *info)
 
 	arizona_extcon_do_magic(info, 0);
 
+	pm_runtime_mark_last_busy(info->dev);
 	pm_runtime_put_autosuspend(info->dev);
 }
 EXPORT_SYMBOL_GPL(arizona_hpdet_stop);
@@ -871,7 +872,7 @@ int arizona_micd_start(struct arizona_extcon_info *info)
 	int ret;
 
 	/* Microphone detection can't use idle mode */
-	pm_runtime_get(info->dev);
+	pm_runtime_get_sync(info->dev);
 
 	ret = regulator_enable(info->micvdd);
 	if (ret != 0) {
-- 
1.7.9.5

